backend/main/server/app.go

```diff
@@ -752,8 +775,7 @@ func (a *App) createProposal(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// get latest snapshotted blockheight
-	snapshot, err := a.SnapshotClient.GetLatestSnapshot()
+	snapshot, err := a.SnapshotClient.GetLatestFlowSnapshot()
```

Comment:
This call is here to set `p.Block_height`, which works for any FlowToken based strategy, but now that we are triggering a snapshot in certain cases, we need to use the block height that the triggered snapshot is going to be for.

So maybe we do need the `GetLatestSnapshot` call down below, or if you can find a way to return the blockheight being snapshotted from the `TakeSnapshot` API call, that would be the best solution with the least number of network requests.